beginner_source/onnx/onnx_diagnostics_tutorial.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+
+"""
+`Introduction to ONNX <intro_onnx.html>`_ ||
+`Export a PyTorch model to ONNX <export_simple_model_to_onnx_tutorial.html>`_ ||
+`Introduction to ONNX Registry <onnx_registry_tutorial.html>`_ ||
+**Introduction to ONNX Diagnostics**
+
+Introduction to ONNX Diagnostics
+================================
+
+**Author:** Bowen Bao (bowbao@microsoft.com)
+"""
+
+###############################################################################
+# Overview
+# --------
+#
+# Welcome to this tutorial on ONNX Diagnostics. ONNX diagnostics goes beyond regular logs through the adoption of
+# `Static Analysis Results Interchange Format (aka SARIF) <https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html>`__
+# to help users debug and improve their model using a GUI, such as
+# Visual Studio Code's `SARIF Viewer <https://marketplace.visualstudio.com/items?itemName=MS-SarifVSCode.sarif-viewer>`_,
+# or `online SARIF Viewer <https://microsoft.github.io/sarif-web-component/>`_, etc.
+# Each diagnostic rule is documented at `ONNX Diagnostic Rules <https://pytorch.org/docs/main/onnx_dynamo.html#diagnosing-issues-with-sarif>`_.
+#
+# Benefits of ONNX Diagnostics include:
+#
+# - Machine-parsable `Static Analysis Results Interchange Format (SARIF) <https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html>`__ output.
+# - A clearer and structured approach for adding and tracking diagnostic rules.
+# - Serving as a foundation for future enhancements leveraging the diagnostics.
+#
+# This tutorial will guide you on:
+#
+# 1. Setting up the SARIF Viewer.
+# 2. Generating a SARIF diagnostic log from ONNX export.
+# 3. Loading and exploring SARIF diagnostics in SARIF Viewer.
+# 4. Increasing diagnostic verbosity for in-depth debugging.
+#
+
+######################################################################
+# Setting up the SARIF Viewer
+# ---------------------------
+#
+# Various tools can visualize SARIF files, including the`SARIF Viewer <https://marketplace.visualstudio.com/items?itemName=MS-SarifVSCode.sarif-viewer>`_
+# extension for VSCode and the `online SARIF Viewer <https://microsoft.github.io/sarif-web-component/>`_.
+# This tutorial will focus on the VSCode SARIF Viewer extension.
+#
+# To install:
+#
+# - Open VSCode.
+# - Search for "SARIF Viewer" in the extension marketplace.
+# - Install the extension.
+#
+# .. image:: ../_static/img/onnx/diagnostics_install_sarif_viewer.png
+#
+
+######################################################################
+# Generating a SARIF diagnostic log from ONNX export
+# --------------------------------------------------
+#
+# Let's begin by exporting a simple model:
```

Comment:
Maybe mention that SARIF is enabled by default? Can it be disabled by users in production?